### PR TITLE
refactor: Replaced deprecated Gtk::StockID in Widget Interpolation

### PR DESCRIPTION
**Pull request by ice0**
Before:
Screenshot_54

After:
Screenshot_55

### CODE REVIEW

File: synfig-studio/src/gui/widgets/widget_interpolation.cpp [hunk 1]
```diff
@@ -42,34 +42,47 @@
...
+	auto interpolation = ParamDesc("interpolation")
+		.set_hint("enum")
+		.add_enum_value(INTERPOLATION_CLAMPED,"clamped",_("Clamped"))
+		.add_enum_value(INTERPOLATION_TCB,"auto",_("TCB"))
+		.add_enum_value(INTERPOLATION_CONSTANT,"constant",_("Constant"))
+		.add_enum_value(INTERPOLATION_HALT,"ease",
+						side == SIDE_BOTH? _("Ease In/Out") : (
+								side == SIDE_BEFORE? _("Ease In") :
+								_("Ease Out")))
+		.add_enum_value(INTERPOLATION_LINEAR,"linear",_("Linear"));
```
Comment (rodolforg) [inline synfig-studio/src/gui/widgets/widget_interpolation.cpp:81]: As you are rewriting it, let's add space after comma?

Comment (ice0) [reply synfig-studio/src/gui/widgets/widget_interpolation.cpp]: Done. Please check. I also aligned them, but I can revert to one-space only.

File: synfig-studio/src/gui/widgets/widget_interpolation.cpp [hunk 2]
```diff
@@ -42,34 +42,46 @@
...
+		.add_enum_value(INTERPOLATION_TCB,      "auto",     _("TCB"))
+		.add_enum_value(INTERPOLATION_CONSTANT, "constant", _("Constant"))
+		.add_enum_value(INTERPOLATION_HALT,     "ease",
+						side == SIDE_BOTH? _("Ease In/Out") : (
+						side == SIDE_BEFORE? _("Ease In") : _("Ease Out")))
+		.add_enum_value(INTERPOLATION_LINEAR,   "linear",   _("Linear"));
+
+	set_param_desc(interpolation);
+	int i = 0;
+	for (auto& item : interpolation.get_enum_list()) {
```
Comment (rodolforg) [inline synfig-studio/src/gui/widgets/widget_interpolation.cpp:84]: `const auto& item`?

Comment (rodolforg) [inline synfig-studio/src/gui/widgets/widget_interpolation.cpp:84]: `const auto& item`?

File: synfig-studio/src/gui/widgets/widget_interpolation.cpp [hunk 3]
```diff
@@ -42,34 +42,46 @@
...
-	const char* interpolation_icons[num_interpolations] = {
-		"synfig-interpolation_type_clamped",
-		"synfig-interpolation_type_tcb",
-		"synfig-interpolation_type_const",
-		"synfig-interpolation_type_ease",
-		"synfig-interpolation_type_linear"
-	};
-	for (auto i = 0; i < num_interpolations; ++i)
-		set_icon(i, Gtk::Button().render_icon_pixbuf(Gtk::StockID(interpolation_icons[i]),Gtk::ICON_SIZE_MENU));
+	auto interpolation = ParamDesc("interpolation")
```
Comment (rodolforg) [inline synfig-studio/src/gui/widgets/widget_interpolation.cpp:72]: `const auto`?

Comment (rodolforg) [inline synfig-studio/src/gui/widgets/widget_interpolation.cpp:72]: `const auto`?

File: synfig-studio/src/gui/widgets/widget_interpolation.cpp (hunk 2 above)
Comment (ice0) [reply synfig-studio/src/gui/widgets/widget_interpolation.cpp]: Done.

File: synfig-studio/src/gui/widgets/widget_interpolation.cpp (hunk 3 above)
Comment (ice0) [reply synfig-studio/src/gui/widgets/widget_interpolation.cpp]: Done.

File: synfig-studio/src/gui/widgets/widget_interpolation.cpp [hunk 4]
```diff
@@ -42,34 +42,46 @@
 using namespace synfig;
 using namespace studio;
 
-Widget_Interpolation::Widget_Interpolation(Side side)
+//TODO(ice0): duplicated code
```
Comment (rodolforg) [inline synfig-studio/src/gui/widgets/widget_interpolation.cpp:45]: Maybe #include iconcontroller.h here?

Comment (ice0) [reply synfig-studio/src/gui/widgets/widget_interpolation.cpp]: Done.

**Review by rodolforg — APPROVED** [review]

**Pull request merged**